File: LeetCode/Medium/Count Primes/Go/count_primes.go
package main

import "fmt"

func sieve(n int) []bool {
	is_prime := make([]bool, n)

	for i := 2; i < n; i++ {
		is_prime[i] = true
	}

	for i := 2; i*i < n; i++ {
		if !is_prime[i] {
			continue
		}
		for j := i * i; j < n; j += i {
			is_prime[j] = false
		}
	}
	return is_prime
}

func count_prime(n int) int {
	cnt := 0
	is_prime := sieve(n)

	for i := 2; i < n; i++ {
		if is_prime[i] {
			cnt++
		}
	}
	return cnt
}

func main() {
	fmt.Println(count_prime(10))
}

File: LeetCode/Medium/Count Primes/Cpp/count_primes.cc
#include <iostream>
#include <vector>

typedef std::vector<bool> vb; 
typedef long long ll;

vb sieve(ll n) {
    vb is_prime(n);
    for (ll i=2; i<n; i++) is_prime[i] = true;
    for (ll i=2; i<n; i++) {
        if (!is_prime[i]) continue;
        for (ll j= i*i; j<n; j+=i) {
            is_prime[j] = false;
        }
    }
    return is_prime;
}

int count_primes(int n) {
    vb is_prime = sieve(n);
    int cnt = 0;
    for (int i=2; i<n; i++) if (is_prime[i]) cnt++;
    return cnt;
}

int main() {
    std::cout<<count_primes(10)<<"\n";
}
